client/pkg/transport/listener.go

```diff
@@ -501,12 +513,88 @@ func (info TLSInfo) ServerConfig() (*tls.Config, error) {
 	// and in TLS 1.3, cipher suites are not configurable
 	// setting Max TLS version to TLS 1.2 for go 1.13
 	cfg.MaxVersion = tls.VersionTLS12
-
 	return cfg, nil
 }
 
+// baseConfig is called on initial TLS handshake start.
+//
+// Previously,
```

Comment:
Looks like you are describing the fix, however comments should reflect current state of the code. Could you possibly add this information to commit message.